workflows/phaseimpute/main.nf

```diff
@@ -396,6 +402,46 @@ workflow PHASEIMPUTE {
             ch_input_validate = ch_input_validate.mix(CONCAT_QUILT.out.vcf_tbi)
         }
 
+        if (params.tools.split(',').contains("minimac4")) {
+            log.info("Impute with MINIMAC4")
+
+            // Prepare VCF input channel with clean metadata
+            ch_vcf_clean = ch_input_type.vcf
+                .map { meta, vcf, index -> 
+                    [[id: meta.id], vcf, index]
+                }
+
+            // Extract chromosome information from VCF
+            VCFCHREXTRACT_MINIMAC4(ch_vcf_clean.map { meta, vcf, index -> [meta, vcf] })
+            
+            // Create per-chromosome channels for MINIMAC4
+            ch_minimac4_input = VCFCHREXTRACT_MINIMAC4.out.chr
+                .join(ch_vcf_clean, by: 0)
+                .map { meta, chr_file, vcf, index ->
+                    chr_file.readLines().collect { chr ->
+                        [[id: meta.id, chr: chr], vcf, index]
+                    }
+                }
+                .flatten()
+                .collate(3)
```

Comment:
I'm not sure about this part.
I would prefer to do it as all the other imputation workflow and use the ch_region channel.
Normally this should automatically be subsampled to only use the chromosome present in all the files.